File: backend/Models/Veiling.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{

    /// <summary>
    /// Maakt een model aan voor Veiling Gegevens met de benodigde gegevens ervoor
    /// </summary>
    [Table("veiling")]

    public class Veiling
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int VeilingId { get; set; }
        
        [Required]
        public string Naam { get; set; } = string.Empty;
        
        public string Beschrijving { get; set; } = string.Empty;
        
        public string Image {get; set; } = string.Empty;
        
        public DateTimeOffset Starttijd { get; set; }
        
        public DateTimeOffset? Eindtijd { get; set; }

        /// <summary>
        /// De duur per productcyclus in seconden (1-30).
        /// </summary>
        [Required]
        [Range(1, 30)]
        public int VeilingDuurInSeconden { get; set; } = 10;

        public int VeilingMeesterId { get; set; }

        public int LocatieId { get; set; }
    }
}

File: frontend/types/veiling.ts
import { Product } from './product';

export interface Veiling {
  veilingId: number;
  naam: string;
  description?: string; // Backend has 'beschrijving', and sometimes UI uses 'description'
  beschrijving: string;
  image: string;
  starttijd: string;
  eindtijd?: string;
  veilingDuurInSeconden: number;
  locatieId: number;
  veilingMeesterId: number;
  producten?: Product[];
  hasUnfinishedProducts?: boolean; // Computed by API - true if auction has products not yet auctioned
}

export interface VeilingDisplayProps {
  veilingen: Veiling[];
}
